frontend/src/metabase/static-viz/components/XYChart/utils/series.unit.spec.ts

```diff
@@ -165,4 +164,66 @@ describe("calculateStackedItems", () => {
       ],
     ]);
   });
+
+  it("calculates stacked items separating positive and negative values even when data has gaps #20752", () => {
+    const stackedSeriesWithGaps: Series[] = [
+      {
+        name: "series 1",
+        color: "#509ee3",
+        yAxisPosition: "left",
+        type: "area",
+        data: [
+          ["2020-10-18", 10],
+          ["2020-10-19", -10],
+        ],
+      },
+      {
+        name: "series 2",
+        color: "#a989c5",
+        yAxisPosition: "left",
+        type: "area",
+        data: [
+          ["2020-10-19", -20],
```

Comment:
Note that the second series purposely start on day 19 rather than day 18 to make sure our code can still calculate stacked series correctly.